pkg/transport/fetchidentity.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package transport
+
+import (
+	"context"
+	"net"
+
+	"github.com/zeebo/errs"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/peertls"
+)
+
+// TODO: check whether this would be problematic with concurrency or should we thread the authinfo through context?
+type handshakeCapture struct {
+	credentials.TransportCredentials
+	AuthInfo credentials.AuthInfo
+}
+
+// ClientHandshake does the authentication handshake specified by the corresponding
+// authentication protocol on rawConn for clients. It returns the authenticated
+// connection and the corresponding auth information about the connection.
+func (capture *handshakeCapture) ClientHandshake(ctx context.Context, s string, conn net.Conn) (net.Conn, credentials.AuthInfo, error) {
+	conn, auth, err := capture.TransportCredentials.ClientHandshake(ctx, s, conn)
+	capture.AuthInfo = auth
+	return conn, auth, err
```

Comment:
nit - does the error need to be wrapped?

Reply:
here, no, because it's wrapping and we don't know how the error is handled internally, by wrapping we might lose important interface methods.